docs/release/2024q1.md

```diff
@@ -0,0 +1,161 @@
+## 2024-04-01: 2024q1 release
+
+This release covers everything from 2024-01-01 and has been tested with avr-gcc
+v12.2.0 from Upstream and arm-none-eabi-gcc v12.2.1 from xpack.
+
+Breaking changes:
+
+- TinyUSB v0.16 implements a DFU class in addition to a DFU *runtime* class.
+  Therefore the `modm:tinyusb:config:device.dfu` option was renamed to
+  `device.dfu_rt` to align with the TinyUSB naming.
+- TinyUSB v0.16 improves support for using two USB ports at the same time. A
+  differnt port can now be chosen for device and host classes. In addition, the
+  HS peripheral may be operated in FS mode, therefore the `modm:tinyusb:speed`
+  option was renamed to `modm:tinyusb:max-speed` since it now has a different
+  meaning.
+- Remove the separate `hosted-rpi` Raspberry Pi target in favor of plain
+  `hosted-linux`. Its GPIO implementation based on wiringPi is deprecated.
+
+Features:
+
+- Add USB High-Speed support to all development boards with ULPI interface.
+- Catch accidental use of Newlib's printf implementation.
```

Comment:
Feature or fix?

Reply:
The detection of this problem is a new feature, there is no fix other than to include `modm:printf`, but this needs to be decided by the user. So it's just better error reporting basically.